ember-client/shared/src/main/scala/org/http4s/ember/client/internal/WebSocketHelpers.scala

```diff
@@ -79,15 +102,26 @@ private[client] object WebSocketHelpers {
           .withUri(wsRequest.uri)
           .withHeaders(wsRequest.headers)
           .withMethod(Method.GET)
+
         for {
           socketResource <- getSocket(emberClient, httpWSRequest)
           socketOption <- socketResource
+          socket = socketOption.get
+
+          clientReceiveQueue <- Queue.synchronous[F, WebSocketFrame].toResource
+          clientSendQueue <- Queue.synchronous[F, WebSocketFrame].toResource
         } yield new WSConnection[F] {
-          def receive: F[Option[org.http4s.client.websocket.WSFrame]] = ???
-          def send(wsf: org.http4s.client.websocket.WSFrame): F[Unit] = ???
-          def sendMany[G[_], A <: org.http4s.client.websocket.WSFrame](wsfs: G[A])(implicit
+          def receive: F[Option[WSFrame]] = for {
+            _ <- socket.reads.through(decodeFrames(true)).evalTap(clientReceiveQueue.offer(_)).compile.drain
+            frame <- clientReceiveQueue.take.map(toWSFrame(_).some)
+          } yield frame
+          def send(wsf: WSFrame): F[Unit] = for {
+            _ <- toWebSocketFrame(wsf).map(clientSendQueue.offer(_))
+            _ <- clientSendQueue.take.map(f => frameToBytes(f, true).traverse_(c => socket.write(c)))
```

Comment:
Same with this one, I think you might need to do `foreverM.background` so that it loops forever.